pkgs/by-name/fa/fawltydeps/package.nix

```
@@ -0,0 +1,57 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  writableTmpDirAsHomeHook,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "fawltydeps";
+  version = "0.20.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "tweag";
+    repo = "FawltyDeps";
+    tag = "v${version}";
+    hash = "sha256-RGwCi4SD0khuOZXcR9Leh9WtRautnlJIfuLBnosyUgk=";
+  };
+
+  build-system = with python3Packages; [ poetry-core ];
+
+  dependencies = with python3Packages; [
+    pyyaml
+    importlib-metadata
+    isort
+    pip-requirements-parser
+    pydantic
+  ];
+
+  nativeCheckInputs =
+    [
+      writableTmpDirAsHomeHook
+    ]
+    ++ (with python3Packages; [
+      pytestCheckHook
+      hypothesis
+    ]);
+
+  disabledTestPaths = [
+    # Try to access files.pythonhosted.org
+    "tests/test_install_deps.py"
+    "tests/test_local_env.py"
+    "tests/test_packages.py"
+    "tests/test_resolver.py"
+    "tests/test_settings.py"
+    "tests/test_traverse_project.py"
+    "tests/test_types.py"
+  ];
```

I played around with the nix expression a bit, and I'm able to make it build with these changes:

```suggestion
  nativeCheckInputs = with python3Packages; [
    pytestCheckHook
    hypothesis
  ];

  disabledTestPaths = [
    # Disable tests that require network
    "tests/test_install_deps.py"
    "tests/test_resolver.py"
  ];

  pythonImportsCheck = [ "fawltydeps" ];
```

---

Adding `writableTmpDirAsHomeHook` to supress this warning:

```
../../nix/store/27q353293bpw41z784qz6cjjwp4mbrs6-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/cacheprovider.py:475
  /nix/store/27q353293bpw41z784qz6cjjwp4mbrs6-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/cacheprovider.py:475: PytestCacheWarning: could not create cache path /homeless-shelter/.cache/pytest/v/cache/nodeids: [Errno 13] Permission denied: '/homeless-shelter'
    config.cache.set("cache/nodeids", sorted(self.cached_nodeids))

../../nix/store/27q353293bpw41z784qz6cjjwp4mbrs6-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/stepwise.py:51
  /nix/store/27q353293bpw41z784qz6cjjwp4mbrs6-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/stepwise.py:51: PytestCacheWarning: could not create cache path /homeless-shelter/.cache/pytest/v/cache/stepwise: [Errno 13] Permission denied: '/homeless-shelter'
    session.config.cache.set(STEPWISE_CACHE_DIR, [])

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```